### PR TITLE
[Bugfix ] fix bailing_moe_linear

### PR DESCRIPTION
**Pull request by ghphotoframe**
## Purpose
1. fix decode idx err vllm v1 reorders batches as decode-first, but _decode_infer was ported from SGLang which uses prefill-first ordering. In mixed batches, this caused decode requests to read q/k/v from wrong positions and update wrong KV cache state slots, corrupting the recurrent state and leading to degenerate output (repeated newlines) in long-form generation.

    The bug is silently masked in decode-only batches (num_prefill_tokens=0) but triggers frequently under high concurrency when prefill and decode requests are scheduled together.

    Fix: use q_start=0/q_end=num_decode_tokens and slot_start=0/slot_end=num_decodes to correctly slice the decode portion.

2. make BailingMoELinearAttention a PluggableLayer Convert BailingMoELinearAttention from inheriting (nn.Module, MambaBase) to (PluggableLayer, MambaBase) and register it with 
`@PluggableLayer.register("bailing_moe_linear_attention").`

    This enables out-of-tree (OOT) backends to transparently replace the entire class at instantiation time via PluggableLayer.register_oot, instead of relying on fragile monkey-patching of individual methods.
    - Add PluggableLayer import to bailing_moe_linear.py
    - Change base class from nn.Module to PluggableLayer (which is itself an nn.Module subclass, so MRO remains compatible with MambaBase)
    - Add @PluggableLayer.register decorator with name "bailing_moe_linear_attention"
    - No functional change for existing GPU/CPU backends; the default forward path is unchanged

4. remove the restriction on float32 cache types in Mamba utilities 
5. ensures proper prefixing for the language model head
6. Remove the explicit float32 dtype specification in `get_rope` to let it use the model's default dtype. This change:
    - Fixes dtype compatibility errors on certain platforms
    - Aligns with the behavior of other models in vLLM
    - Matches the RoPE dtype used in SGLang's bailing adapter

## Test Plan

- Run GSM8K benchmark under high concurrency (mixed prefill/decode workloads)
- Compare accuracy before and after the fix
Test command:
```bash
# High concurrency GSM8K evaluation
source port.sh
evalscope eval \
  --model auto \
  --model-id auto \
  --datasets gsm8k \
  --dataset-dir /datasets/gsm8k\
  --chat-template true \
  --api-url http://localhost:${PORT}/v1\
  --eval-batch-size 32
```

## Test Result
Verified: GSM8K accuracy recovers from ~84% to ~96% under high concurrency.

Signed-off-by: ghphotoframe <854746559@qq.com>

### CODE REVIEW

**Comment by ghphotoframe** [comment]
Hi @ZJY0516 , thanks for verifying the changes! 
It looks like we still need an official **approval**  to trigger the remaining CI checks.  
Would you mind approving the PR when you get a chance? Thanks again!

**Review by ZJY0516 — APPROVED** [review]
